packages/govuk-frontend/src/govuk/components/input/template.jsdom.test.js

```diff
@@ -0,0 +1,496 @@
+    examples = await getExamples('input')
+  })
+
+  describe('default example', () => {
+    let $component
+    let $label
+
+    beforeAll(() => {
+      document.body.innerHTML = render('input', examples.default)
+      $component = document.querySelector('.govuk-input')
+      $label = document.querySelector('.govuk-label')
+    })
+
+    it('sets the `id` attribute based on the `id` option', () => {
+      expect($component).toHaveAttribute('id', 'input-example')
+    })
+
+    it('sets the `name` attribute based on the `name` option', () => {
+      expect($component).toHaveAttribute('name', 'test-name')
+    })
+
+    it('sets the `type` attribute to a default value of "text"', () => {
+      expect($component).toHaveAttribute('type', 'text')
+    })
+
+    it('includes a form group wrapper', () => {
+      const $formGroup = document.querySelector('.govuk-form-group')
+
+      expect($formGroup).toBeInTheDocument()
+      expect($formGroup).toContainElement($component)
+    })
+
+    it('includes a label', () => {
+      expect($label).toBeInTheDocument()
+    })
+
+    it('the input is named by the label', () => {
+      expect($component).toHaveAccessibleName($label.textContent.trim())
+    })
+
+    it('does not include the input wrapper', () => {
+      const $wrapper = document.querySelector(
+        '.govuk-form-group > .govuk-input__wrapper'
+      )
+
+      expect($wrapper).toBeNull()
+    })
+
+    it.each([
+      'autocapitalize',
+      'autocomplete',
+      'disabled',
+      'inputmode',
+      'pattern',
+      'spellcheck'
+    ])('does not set the `%s` attribute', (attribute) => {
+      expect($component).not.toHaveAttribute(attribute)
+    })
+  })
+
+  describe('custom options', () => {
+    it('includes additional classes from the `classes` option', () => {
+      document.body.innerHTML = render('input', examples.classes)
+
+      const $component = document.querySelector('.govuk-input')
+      expect($component).toHaveClass('app-input--custom-modifier')
+    })
+
+    it('sets the `type` attribute based on the `type` option', () => {
+      document.body.innerHTML = render('input', examples['custom type'])
+
+      const $component = document.querySelector('.govuk-input')
+      expect($component).toHaveAttribute('type', 'number')
+    })
+
+    it('sets the `pattern` attribute based on the `pattern` option', () => {
+      document.body.innerHTML = render(
+        'input',
+        examples['with pattern attribute']
+      )
+
+      const $component = document.querySelector('.govuk-input')
+      expect($component).toHaveAttribute('pattern', '[0-9]*')
+    })
+
+    it('sets the `value` attribute based on the `value` option', () => {
+      document.body.innerHTML = render('input', examples.value)
+
+      const $component = document.querySelector('.govuk-input')
+      expect($component).toHaveValue('QQ 12 34 56 C')
+    })
+
+    it('sets the `value` attribute based on a `value` of "0"', () => {
```

Comment:
**nitpick** I think the issue was with a `0` value (falsey, made the attribute not render) rather than a string of `"0"` (truthey).

```suggestion
    it('sets the `value` attribute based on a `value` of `0`', () => {
```

If there's an issue with both, we should probably have two separate tests 😊